### PR TITLE
participation rewards update

### PR DESCRIPTION
**Pull request by zanicar**
- added zone allocation state
- zero allocations on callback
- merged zone proportions and allocations (less looping required)
- devtest debug output removed
- removed unnecessary params from functions

### CODE REVIEW

File: x/participationrewards/keeper/callbacks.go [hunk 1]
```diff
@@ -89,5 +85,9 @@ func RewardsCallback(k Keeper, ctx sdk.Context, response []byte, query icqtypes.
 		k.icsKeeper.SetIntent(ctx, zone, di, true)
 	}
 
+	// set zone ValidatorSelectionAllocation to zero
+	zone.ValidatorSelectionAllocation = sdk.NewCoins(sdk.NewCoin("uqck", sdk.ZeroInt()))
```
Comment (joe-bowman) [inline x/participationrewards/keeper/callbacks.go:89]: whilst this always will be `uqck`, it's probably correct to fetch `k.BondDenom(ctx)` from the staking keeper.

File: x/participationrewards/keeper/rewards_validatorSelection.go [hunk 2]
```diff
@@ -292,7 +285,7 @@ func (k Keeper) calcUserAllocations(
 		return userAllocations, nil
 	}
 
-	tokensPerPoint := za.AmountOf("uqck").ToDec().Quo(sum)
+	tokensPerPoint := zone.ValidatorSelectionAllocation.AmountOfNoDenomValidation("uqck").ToDec().Quo(sum)
```
Comment (joe-bowman) [inline x/participationrewards/keeper/rewards_validatorSelection.go:288]: also here :)

**Review by joe-bowman — CHANGES_REQUESTED** [review]
lgtm - but if we can replace the explicit `uqck` references with a `stakingKeeper.BondDenom(ctx)` call, then we'll avoid a footgun later on.

File: x/participationrewards/keeper/rewards_validatorSelection.go (hunk 2 above)
Comment (zanicar) [reply x/participationrewards/keeper/rewards_validatorSelection.go]: done

**Pull request merged**